### PR TITLE
Feature Collection End point and Integration

### PR DESCRIPTION
**Pull request by SyedSibtainRazvi**
# Etiquettes of raising a PR

PR reviews are a way to request peers to give feedback on the code that is going
to the mainline. This means, that PR should be in a reviewable state. Please ensure
that the following checkboxes are ticked. If any of them are not ticked, then
the PR is not ready for review

- [x] Did you test changes locally?
- [x] Did you check PR contents after opening the PR

# Related issue

- 

# Functionalities added or changed

- 

# Additional info
- https://strapi.io/blog/how-to-create-a-custom-api-endpoint-in-strapi
# Checklist

- [ ] Added/Updated Story
- [ ] Tests
- [ ] Translations
- [ ] Documentation
- [x] Uploaded screenshots
- [x] Verified on preview URL once

# Screenshots

## Page(s) if applicable

|   Device    |            Design             |           Implementation            |
| :---------: | :---------------------------: | :---------------------------------: |
| **Desktop** | ** Figma design screenshot ** |Screenshot 2023-04-06 at 10 58 42 PM|

### CODE REVIEW

File: admin/src/components/ClearCache.jsx [hunk 1]
```diff
@@ -1,18 +1,19 @@
...
+import React, { useState, useEffect } from "react";
 import { Box } from "@strapi/design-system/Box";
 import { Button } from "@strapi/design-system/Button";
 import { Stack } from "@strapi/design-system/Stack";
 import { Select, Option } from "@strapi/design-system/Select";
-import { refreshCache } from "../utils/api";
+import { getCollectionTypes, refreshCache } from "../utils/api";
 
-import { options } from "../utils/options";
+// import { options } from "../utils/options";
```
Comment (SujayPrabhu96) [inline admin/src/components/ClearCache.jsx]: remove this. Also delete the file utils/options file

File: admin/src/components/ClearCache.jsx [hunk 2]
```diff
@@ -1,18 +1,19 @@
...
 import { Alert } from "./Alert";
 
 export const ClearCache = () => {
   const [loading, setLoading] = useState(false);
   const [showAlert, setShowAlert] = useState(false);
   const [variant, setVariant] = useState("success");
   const [collection, setCollection] = useState("");
+  const [options, setoptions] = useState([]);
```
Comment (SujayPrabhu96) [inline admin/src/components/ClearCache.jsx]:
```suggestion
  const [options, setOptions] = useState([]);
```

File: admin/src/utils/api.js [hunk 3]
```diff
@@ -29,7 +29,22 @@ export const refreshCollectionCache = async (query) => {
     const response = await axios({
       url: "stellate/cache/refreshcollection",
       method: "POST",
-      data: query,
+      data: {
+        collection: query,
```
Comment (SujayPrabhu96) [inline admin/src/utils/api.js]: Lets rename `query` to collection
```suggestion
        collection,
```

**Review by SujayPrabhu96 — CHANGES_REQUESTED** [review]

File: admin/src/components/ClearCache.jsx (hunk 1 above)
Comment (SyedSibtainRazvi) [reply admin/src/components/ClearCache.jsx]: Removed

File: admin/src/components/ClearCache.jsx (hunk 2 above)
Comment (SyedSibtainRazvi) [reply admin/src/components/ClearCache.jsx]: Updated

File: admin/src/utils/api.js (hunk 3 above)
Comment (SyedSibtainRazvi) [reply admin/src/utils/api.js]: updated

**Review by SujayPrabhu96 — APPROVED** [review]

**Pull request merged**